facade/src/test/java/org/jboss/pnc/facade/providers/BuildConfigProviderTest.java

```diff
@@ -0,0 +1,258 @@
+        assertThat(buildConfiguration.getName()).isEqualTo(bc.getName());
+    }
+
+    @Test
+    public void testGetAll() {
+        Page<org.jboss.pnc.dto.BuildConfiguration> all = provider.getAll(0, 10, null, null);
+
+        assertThat(all.getContent())
+                .hasSize(5)
+                .haveExactly(1, new Condition<>(e -> e.getName().equals(bc.getName()), "BC Present"));
+    }
+
+    @Test
+    public void testStore() {
+        final String name = "NewHire";
+        final String repoId = "12";
+        final String envId = "12";
+        final String projId = "12";
+
+        org.jboss.pnc.dto.BuildConfiguration buildConfiguration = org.jboss.pnc.dto.BuildConfiguration.builder()
+                .name(name)
+                .buildType(BuildType.MVN)
+                .project(org.jboss.pnc.dto.Project.builder().id(projId).build())
+                .environment(Environment.builder().id(envId).build())
+                .scmRepository(SCMRepository.builder().id(repoId).build())
+                .build();
+
+        org.jboss.pnc.dto.BuildConfiguration stored = provider.store(buildConfiguration);
+
+        assertThat(stored).isNotNull();
+        assertThat(stored.getId()).isNotNull();
+        assertThat(stored.getName()).isEqualTo(name);
+        assertThat(stored.getProject().getId()).isEqualTo(projId);
+        assertThat(stored.getBuildType()).isEqualTo(BuildType.MVN);
+        assertThat(stored.getEnvironment().getId()).isEqualTo(envId);
+        assertThat(stored.getScmRepository().getId()).isEqualTo(repoId);
+
+    }
+
+    //FIXME unignore after NCL-5114 is implemented
+    @Test
+    @Ignore
+    public void testDeleteWithArchival() {
+        //With
+        assertThat(bc.getActive()).isTrue();
+
+        //When
+        provider.delete(bc.getId().toString());
+
+        //Then
+        org.jboss.pnc.dto.BuildConfiguration archived = provider.getSpecific(bc.getId().toString());
+        assertThat(archived).isNotNull();
+        assertThat(archived.isArchived()).isTrue();
+    }
+
+    @Test
+    public void testAddDependency() {
+        //With
+        org.jboss.pnc.dto.BuildConfiguration dependency = provider.getSpecific("2"); //BC(name: "First!")
+        assertThat(dependency.getDependencies()).isNullOrEmpty();
+
+        //When
+        provider.addDependency(dependency.getId(), bc.getId().toString());
+
+        //Then
+        org.jboss.pnc.dto.BuildConfiguration refreshed = provider.getSpecific(dependency.getId());
+        assertThat(refreshed.getDependencies())
+                .anySatisfy(config -> assertThat(config.getId()).isEqualTo(bc.getId().toString()));
+    }
+
+    @Test
+    public void testRemoveDependency() {
+        //With
+        org.jboss.pnc.dto.BuildConfiguration dependency = provider.getSpecific("2"); //BC(name: "First!")
+        org.jboss.pnc.dto.BuildConfiguration dependant = provider.getSpecific("3"); //BC(name: "Second!!")
```

Comment:
not confusing at all ;-)

Reply:
any ideas for improvement? Change naming to child and parent?